openshift/templates/iiw-book-service/iiw-book-service-deploy.yaml

```diff
@@ -4,50 +4,104 @@ apiVersion: v1
+#     labels:
+#       name: ${NAME}
+#       network: ${NAME}
+#       app: ${APP_GROUP}
+#       app-group: ${APP_GROUP}
+#       env: ${TAG_NAME}
+#   spec:
+#     description: |
+#       Define the network parameters for accessing smtp server
+#     entries:
+#       - ${SMTP_EMAIL_HOST}
+#     servicePorts:
+#       - tcp/25
+# - kind: NetworkSecurityPolicy
+#   apiVersion: security.devops.gov.bc.ca/v1alpha1
+#   metadata:
+#     name: ${NAME}
+#     labels:
+#       name: ${NAME}
+#       env: ${TAG_NAME}
+#       app: ${APP_GROUP}
+#       app-group: ${APP_GROUP}
+#   spec:
+#     description: Allow service to access the database and agent
+#     source:
+#       - - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+#         - role=${ROLE}
+#         - app=${APP_GROUP}
+#         - env=${TAG_NAME}
+#     destination:
+#       - - ext:name=${NAME}
+#       - - role=db
+#         - app=${APP_GROUP}
+#         - env=${TAG_NAME}
+#         - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+#       - - role=agent
+#         - app=${APP_GROUP}
+#         - env=${TAG_NAME}
+#         - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+- kind: NetworkPolicy
+  apiVersion: networking.k8s.io/v1
   metadata:
-    name: ${NAME}
-    network: ${NAME}
+    name: ${NAME}-allow-ingress
     labels:
-      name: ${NAME}
-      network: ${NAME}
+      name: ${NAME}-allow-ingress
       app: ${APP_GROUP}
       app-group: ${APP_GROUP}
+      role: ${ROLE}
       env: ${TAG_NAME}
   spec:
     description: |
-      Define the network parameters for accessing smtp server
-    entries:
-      - ${SMTP_EMAIL_HOST}
-    servicePorts:
-      - tcp/25
-- kind: NetworkSecurityPolicy
-  apiVersion: security.devops.gov.bc.ca/v1alpha1
+      Allow traffic from routes into the ${ROLE} container(s).
+    ingress:
+      - from:
+          - namespaceSelector:
+              matchLabels:
+                network.openshift.io/policy-group: ingress
+    podSelector:
+      matchLabels:
+        role: ${ROLE}
+        app: ${APP_GROUP}
+        env: ${TAG_NAME}
+    policyTypes:
+      - Ingress
+- kind: NetworkPolicy
+  apiVersion: networking.k8s.io/v1
   metadata:
     name: ${NAME}
     labels:
       name: ${NAME}
-      env: ${TAG_NAME}
       app: ${APP_GROUP}
-      app-group: ${APP_GROUP}
+      env: ${TAG_NAME}
   spec:
-    description: Allow service to access the database and agent
-    source:
-      - - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
-        - role=${ROLE}
-        - app=${APP_GROUP}
-        - env=${TAG_NAME}
-    destination:
-      - - ext:name=${NAME}
-      - - role=db
-        - app=${APP_GROUP}
-        - env=${TAG_NAME}
-        - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
-      - - role=agent
-        - app=${APP_GROUP}
-        - env=${TAG_NAME}
-        - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+    description: Allow the agent to access the service.
+    ingress:
+      - from:
+        - podSelector:
+            matchLabels:
+              role: agent
```

Comment:
The role of the agent can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.